File: Election Data and Files/Email and Tracking/email_templates.js
// Professional Email Templates for Campaigns

var EMAIL_TEMPLATES = {
    // Template 1: Election Update (Purple Gradient)
    election_update: {
        name: "Election Update",
        theme: "Purple Gradient",
        html: `<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
</head>
<body style="margin: 0; padding: 0; font-family: Arial, sans-serif; background-color: #f5f5f5;">
    <table width="100%" cellpadding="0" cellspacing="0" style="background-color: #f5f5f5; padding: 20px;">
        <tr>
            <td align="center">
                <table width="600" cellpadding="0" cellspacing="0" style="background: white; border-radius: 10px; overflow: hidden; box-shadow: 0 4px 15px rgba(0,0,0,0.1);">
                    <!-- Header -->
                    <tr>
                        <td style="background: linear-gradient(135deg, #667eea 0%, #764ba2 100%); padding: 40px 30px; text-align: center;">
                            <h1 style="color: white; margin: 0; font-size: 28px;">🗳️ Election Update</h1>
                        </td>
                    </tr>
                    <!-- Content -->
                    <tr>
                        <td style="padding: 40px 30px;">
                            <h2 style="color: #333; margin-top: 0;">{{TITLE}}</h2>
                            <p style="color: #555; font-size: 16px; line-height: 1.6;">{{CONTENT}}</p>
                            
                            <!-- CTA Button -->
                            <table width="100%" cellpadding="0" cellspacing="0" style="margin: 30px 0;">
                                <tr>
                                    <td align="center">
                                        <a href="{{CTA_LINK}}" style="display: inline-block; background: linear-gradient(135deg, #667eea 0%, #764ba2 100%); color: white; padding: 15px 40px; text-decoration: none; border-radius: 5px; font-weight: bold; font-size: 16px;">{{CTA_TEXT}}</a>
                                    </td>
                                </tr>
                            </table>
                        </td>
                    </tr>
                    <!-- Footer -->
                    <tr>
                        <td style="background: #f8f9fa; padding: 20px 30px; text-align: center; border-top: 1px solid #eee;">
                            <p style="color: #999; font-size: 12px; margin: 0;">Christian Conservatives Today</p>
                        </td>
                    </tr>
                </table>
            </td>
        </tr>
    </table>
</body>
</html>`
    },

    // Template 2: Urgent Alert (Red Theme)
    urgent_alert: {
        name: "Urgent Alert",
        theme: "Red Alert",
        html: `<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
</head>
<body style="margin: 0; padding: 0; font-family: Arial, sans-serif; background-color: #f5f5f5;">
    <table width="100%" cellpadding="0" cellspacing="0" style="background-color: #f5f5f5; padding: 20px;">
        <tr>
            <td align="center">
                <table width="600" cellpadding="0" cellspacing="0" style="background: white; border-radius: 10px; overflow: hidden; box-shadow: 0 4px 15px rgba(0,0,0,0.1); border: 3px solid #dc3545;">
                    <!-- Header -->
                    <tr>
                        <td style="background: #dc3545; padding: 30px; text-align: center;">
                            <h1 style="color: white; margin: 0; font-size: 24px;">⚠️ URGENT: {{TITLE}}</h1>
                        </td>
                    </tr>
                    <!-- Alert Box -->
                    <tr>
                        <td style="padding: 30px;">
                            <div style="background: #fff3cd; border-left: 4px solid #ffc107; padding: 15px; margin-bottom: 20px;">
                                <p style="margin: 0; color: #856404; font-weight: bold;">Time-Sensitive Action Required</p>
                            </div>
                            <p style="color: #333; font-size: 16px; line-height: 1.6; margin-top: 0;">{{CONTENT}}</p>
                            
                            <!-- CTA Button -->
                            <table width="100%" cellpadding="0" cellspacing="0" style="margin: 25px 0;">
                                <tr>
                                    <td align="center">
                                        <a href="{{CTA_LINK}}" style="display: inline-block; background: #dc3545; color: white; padding: 15px 40px; text-decoration: none; border-radius: 5px; font-weight: bold; font-size: 16px;">{{CTA_TEXT}}</a>
                                    </td>
                                </tr>
                            </table>
                        </td>
                    </tr>
                    <!-- Footer -->
                    <tr>
                        <td style="background: #f8f9fa; padding: 20px 30px; text-align: center; border-top: 1px solid #eee;">
                            <p style="color: #999; font-size: 12px; margin: 0;">Christian Conservatives Today</p>
                        </td>
                    </tr>
                </table>
            </td>
        </tr>
    </table>
</body>
</html>`
    },

    // Template 3: Newsletter (Clean & Professional)
    newsletter: {
        name: "Newsletter",
        theme: "Clean Professional",
        html: `<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
</head>
<body style="margin: 0; padding: 0; font-family: Georgia, serif; background-color: #f5f5f5;">
    <table width="100%" cellpadding="0" cellspacing="0" style="background-color: #f5f5f5; padding: 20px;">
        <tr>
            <td align="center">
                <table width="600" cellpadding="0" cellspacing="0" style="background: white; border-radius: 10px; overflow: hidden; box-shadow: 0 4px 15px rgba(0,0,0,0.1);">
                    <!-- Header -->
                    <tr>
                        <td style="background: #2c3e50; padding: 30px; text-align: center;">
                            <h1 style="color: white; margin: 0; font-size: 32px; font-weight: normal; letter-spacing: 2px;">CCT NEWSLETTER</h1>
                            <p style="color: #ecf0f1; margin: 10px 0 0 0; font-size: 14px;">{{DATE}}</p>
                        </td>
                    </tr>
                    <!-- Featured Image -->
                    <tr>
                        <td style="padding: 0;">
                            <div style="background: linear-gradient(135deg, #667eea 0%, #764ba2 100%); height: 200px; display: flex; align-items: center; justify-content: center;">
                                <h2 style="color: white; font-size: 28px; margin: 0; padding: 40px; text-align: center;">{{TITLE}}</h2>
                            </div>
                        </td>
                    </tr>
                    <!-- Content -->
                    <tr>
                        <td style="padding: 40px 30px;">
                            <p style="color: #555; font-size: 16px; line-height: 1.8; margin-top: 0;">{{CONTENT}}</p>
                            
                            <!-- CTA Button -->
                            <table width="100%" cellpadding="0" cellspacing="0" style="margin: 30px 0;">
                                <tr>
                                    <td align="center">
                                        <a href="{{CTA_LINK}}" style="display: inline-block; background: #2c3e50; color: white; padding: 15px 40px; text-decoration: none; border-radius: 5px; font-weight: bold; font-size: 16px;">{{CTA_TEXT}}</a>
                                    </td>
                                </tr>
                            </table>
                        </td>
                    </tr>
                    <!-- Footer -->
                    <tr>
                        <td style="background: #34495e; padding: 30px; text-align: center; color: white;">
                            <p style="margin: 0 0 10px 0; font-size: 14px;">Christian Conservatives Today</p>
                            <p style="margin: 0; font-size: 12px; color: #95a5a6;">Equipping believers to vote biblically</p>
                        </td>
                    </tr>
                </table>
            </td>
        </tr>
    </table>
</body>
</html>`
    },

    // Template 4: Prayer & Action (Spiritual Theme)
    prayer_action: {
        name: "Prayer & Action",
        theme: "Spiritual",
        html: `<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
</head>
<body style="margin: 0; padding: 0; font-family: 'Palatino Linotype', serif; background-color: #f5f5f5;">
    <table width="100%" cellpadding="0" cellspacing="0" style="background-color: #f5f5f5; padding: 20px;">
        <tr>
            <td align="center">
                <table width="600" cellpadding="0" cellspacing="0" style="background: white; border-radius: 10px; overflow: hidden; box-shadow: 0 4px 15px rgba(0,0,0,0.1);">
                    <!-- Header -->
                    <tr>
                        <td style="background: linear-gradient(135deg, #8e44ad 0%, #3498db 100%); padding: 40px 30px; text-align: center;">
                            <h1 style="color: white; margin: 0; font-size: 28px;">🙏 Prayer & Action</h1>
                            <p style="color: rgba(255,255,255,0.9); margin: 10px 0 0 0; font-style: italic;">"If my people... will humble themselves and pray"</p>
                        </td>
                    </tr>
                    <!-- Content -->
                    <tr>
                        <td style="padding: 40px 30px;">
                            <h2 style="color: #8e44ad; margin-top: 0; text-align: center;">{{TITLE}}</h2>
                            
                            <!-- Prayer Box -->
                            <div style="background: #f8f9fa; border-left: 4px solid #8e44ad; padding: 20px; margin: 20px 0;">
                                <p style="margin: 0; color: #555; font-size: 15px; line-height: 1.7; font-style: italic;">{{PRAYER}}</p>
                            </div>
                            
                            <p style="color: #555; font-size: 16px; line-height: 1.8;">{{CONTENT}}</p>
                            
                            <!-- Action Steps -->
                            <div style="background: #e8f4f8; padding: 20px; border-radius: 5px; margin: 25px 0;">
                                <h3 style="color: #3498db; margin-top: 0; font-size: 18px;">Action Steps:</h3>
                                <p style="color: #555; margin: 0; line-height: 1.7;">{{ACTION_STEPS}}</p>
                            </div>
                            
                            <!-- CTA Button -->
                            <table width="100%" cellpadding="0" cellspacing="0" style="margin: 30px 0;">
                                <tr>
                                    <td align="center">
                                        <a href="{{CTA_LINK}}" style="display: inline-block; background: linear-gradient(135deg, #8e44ad 0%, #3498db 100%); color: white; padding: 15px 40px; text-decoration: none; border-radius: 5px; font-weight: bold; font-size: 16px;">{{CTA_TEXT}}</a>
                                    </td>
                                </tr>
                            </table>
                        </td>
                    </tr>
                    <!-- Footer -->
                    <tr>
                        <td style="background: #f8f9fa; padding: 20px 30px; text-align: center; border-top: 1px solid #eee;">
                            <p style="color: #999; font-size: 12px; margin: 0;">Christian Conservatives Today</p>
                            <p style="color: #bbb; font-size: 11px; margin: 5px 0 0 0; font-style: italic;">"For such a time as this" - Esther 4:14</p>
                        </td>
                    </tr>
                </table>
            </td>
        </tr>
    </table>
</body>
</html>`
    }
};

if (typeof window !== 'undefined') {
    window.EMAIL_TEMPLATES = EMAIL_TEMPLATES;
}

File: newsletter_api/email_templates.py
# Professional Email Templates with Branding

LOGO_URL = "https://christianconservativestoday.com/techcrosslogo.jpg"
PRIMARY_COLOR = "#2c5aa0"
SECONDARY_COLOR = "#8b4513"
ACCENT_COLOR = "#d4af37"

def get_confirmation_email(first_name, confirmation_link):
    return f"""
<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
</head>
<body style="margin: 0; padding: 0; font-family: Arial, sans-serif; background-color: #f4f4f4;">
    <table width="100%" cellpadding="0" cellspacing="0" style="background-color: #f4f4f4; padding: 20px 0;">
        <tr>
            <td align="center">
                <table width="600" cellpadding="0" cellspacing="0" style="background-color: #ffffff; border-radius: 10px; overflow: hidden; box-shadow: 0 4px 6px rgba(0,0,0,0.1);">
                    <!-- Header with gradient -->
                    <tr>
                        <td style="background: linear-gradient(135deg, {PRIMARY_COLOR} 0%, {SECONDARY_COLOR} 100%); padding: 40px 20px; text-align: center;">
                            <img src="{LOGO_URL}" alt="Christian Conservatives Today" style="max-width: 150px; height: auto; margin-bottom: 20px;">
                            <h1 style="color: #ffffff; margin: 0; font-size: 28px;">Confirm Your Subscription</h1>
                        </td>
                    </tr>
                    
                    <!-- Content -->
                    <tr>
                        <td style="padding: 40px 30px;">
                            <p style="font-size: 16px; color: #333333; line-height: 1.6; margin: 0 0 20px 0;">
                                Hi {first_name},
                            </p>
                            <p style="font-size: 16px; color: #333333; line-height: 1.6; margin: 0 0 20px 0;">
                                Thank you for subscribing to <strong>Christian Conservatives Today</strong>! We're excited to have you join our community of believers and truth-seekers.
                            </p>
                            <p style="font-size: 16px; color: #333333; line-height: 1.6; margin: 0 0 30px 0;">
                                Please confirm your email address by clicking the button below:
                            </p>
                            
                            <!-- CTA Button -->
                            <table width="100%" cellpadding="0" cellspacing="0">
                                <tr>
                                    <td align="center" style="padding: 20px 0;">
                                        <a href="{confirmation_link}" style="background-color: {PRIMARY_COLOR}; color: #ffffff; text-decoration: none; padding: 15px 40px; border-radius: 25px; font-size: 18px; font-weight: bold; display: inline-block;">
                                            Confirm My Subscription
                                        </a>
                                    </td>
                                </tr>
                            </table>
                            
                            <p style="font-size: 14px; color: #666666; line-height: 1.6; margin: 30px 0 0 0;">
                                Or copy and paste this link into your browser:<br>
                                <a href="{confirmation_link}" style="color: {PRIMARY_COLOR}; word-break: break-all;">{confirmation_link}</a>
                            </p>
                            
                            <p style="font-size: 14px; color: #999999; line-height: 1.6; margin: 30px 0 0 0; padding-top: 20px; border-top: 1px solid #eeeeee;">
                                This link will expire in 24 hours. If you didn't subscribe to our newsletter, please ignore this email.
                            </p>
                        </td>
                    </tr>
                    
                    <!-- Footer -->
                    <tr>
                        <td style="background-color: #f8f9fa; padding: 30px; text-align: center;">
                            <p style="font-size: 14px; color: #666666; margin: 0 0 10px 0;">
                                <strong>Christian Conservatives Today</strong><br>
                                Where Truth Meets Action
                            </p>
                            <p style="font-size: 12px; color: #999999; margin: 0;">
                                © {datetime.now().year} Christian Conservatives Today. All rights reserved.
                            </p>
                        </td>
                    </tr>
                </table>
            </td>
        </tr>
    </table>
</body>
</html>
"""

def get_welcome_email(first_name):
    return f"""
<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
</head>
<body style="margin: 0; padding: 0; font-family: Arial, sans-serif; background-color: #f4f4f4;">
    <table width="100%" cellpadding="0" cellspacing="0" style="background-color: #f4f4f4; padding: 20px 0;">
        <tr>
            <td align="center">
                <table width="600" cellpadding="0" cellspacing="0" style="background-color: #ffffff; border-radius: 10px; overflow: hidden; box-shadow: 0 4px 6px rgba(0,0,0,0.1);">
                    <!-- Header -->
                    <tr>
                        <td style="background: linear-gradient(135deg, {PRIMARY_COLOR} 0%, {SECONDARY_COLOR} 100%); padding: 40px 20px; text-align: center;">
                            <img src="{LOGO_URL}" alt="Christian Conservatives Today" style="max-width: 150px; height: auto; margin-bottom: 20px;">
                            <h1 style="color: #ffffff; margin: 0; font-size: 28px;">Welcome to Our Community!</h1>
                        </td>
                    </tr>
                    
                    <!-- Content -->
                    <tr>
                        <td style="padding: 40px 30px;">
                            <p style="font-size: 16px; color: #333333; line-height: 1.6; margin: 0 0 20px 0;">
                                Hi {first_name},
                            </p>
                            <p style="font-size: 16px; color: #333333; line-height: 1.6; margin: 0 0 20px 0;">
                                Welcome to <strong>Christian Conservatives Today</strong>! Your subscription is now active, and you'll receive our latest updates, articles, and news.
                            </p>
                            
                            <h2 style="color: {PRIMARY_COLOR}; font-size: 22px; margin: 30px 0 20px 0;">What You'll Receive:</h2>
                            <ul style="font-size: 16px; color: #333333; line-height: 1.8; margin: 0 0 30px 0; padding-left: 20px;">
                                <li>Weekly newsletter with top articles and news</li>
                                <li>Election updates and voter guides for all 50 states</li>
                                <li>Prayer requests and ministry updates</li>
                                <li>Event announcements and community news</li>
                            </ul>
                            
                            <h2 style="color: {PRIMARY_COLOR}; font-size: 22px; margin: 30px 0 20px 0;">Get Started:</h2>
                            
                            <!-- Feature Boxes -->
                            <table width="100%" cellpadding="0" cellspacing="0" style="margin-bottom: 20px;">
                                <tr>
                                    <td style="padding: 20px; background-color: #f8f9fa; border-radius: 8px; border-left: 4px solid {ACCENT_COLOR};">
                                        <h3 style="color: {PRIMARY_COLOR}; font-size: 18px; margin: 0 0 10px 0;">📖 Explore Articles</h3>
                                        <p style="font-size: 14px; color: #666666; margin: 0 0 10px 0;">Read faith-based articles with integrated Bible verses</p>
                                        <a href="https://christianconservativestoday.com/articles.html" style="color: {PRIMARY_COLOR}; text-decoration: none; font-weight: bold;">Browse Articles →</a>
                                    </td>
                                </tr>
                            </table>
                            
                            <table width="100%" cellpadding="0" cellspacing="0" style="margin-bottom: 20px;">
                                <tr>
                                    <td style="padding: 20px; background-color: #f8f9fa; border-radius: 8px; border-left: 4px solid {ACCENT_COLOR};">
                                        <h3 style="color: {PRIMARY_COLOR}; font-size: 18px; margin: 0 0 10px 0;">🎥 Watch Videos</h3>
                                        <p style="font-size: 14px; color: #666666; margin: 0 0 10px 0;">Sermons, teachings, and conservative commentary</p>
                                        <a href="https://christianconservativestoday.com/videos.html" style="color: {PRIMARY_COLOR}; text-decoration: none; font-weight: bold;">Watch Now →</a>
                                    </td>
                                </tr>
                            </table>
                            
                            <table width="100%" cellpadding="0" cellspacing="0" style="margin-bottom: 20px;">
                                <tr>
                                    <td style="padding: 20px; background-color: #f8f9fa; border-radius: 8px; border-left: 4px solid {ACCENT_COLOR};">
                                        <h3 style="color: {PRIMARY_COLOR}; font-size: 18px; margin: 0 0 10px 0;">🗳️ Election Tracking</h3>
                                        <p style="font-size: 14px; color: #666666; margin: 0 0 10px 0;">Voter guides for all 50 states with candidate profiles</p>
                                        <a href="https://christianconservativestoday.com/election-map.html" style="color: {PRIMARY_COLOR}; text-decoration: none; font-weight: bold;">View Map →</a>
                                    </td>
                                </tr>
                            </table>
                            
                            <p style="font-size: 16px; color: #333333; line-height: 1.6; margin: 30px 0 0 0;">
                                God bless you,<br>
                                <strong>The Christian Conservatives Today Team</strong>
                            </p>
                        </td>
                    </tr>
                    
                    <!-- Footer -->
                    <tr>
                        <td style="background-color: #f8f9fa; padding: 30px; text-align: center;">
                            <p style="font-size: 14px; color: #666666; margin: 0 0 10px 0;">
                                <strong>Christian Conservatives Today</strong><br>
                                Where Truth Meets Action
                            </p>
                            <p style="font-size: 12px; color: #999999; margin: 0 0 10px 0;">
                                You're receiving this because you subscribed to our newsletter.
                            </p>
                            <p style="font-size: 12px; color: #999999; margin: 0;">
                                <a href="https://christianconservativestoday.com/unsubscribe.html?email={{{{email}}}}" style="color: #999999;">Unsubscribe</a>
                            </p>
                        </td>
                    </tr>
                </table>
            </td>
        </tr>
    </table>
</body>
</html>
"""

from datetime import datetime
